### PR TITLE
JBIDE-25434 - oc location validation in cdk and openshift wizards

### PR DESCRIPTION
**Pull request by robstryker**
Signed-off-by: Rob Stryker <rob@oxbeef.net>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionComposite.java [hunk 1]
```diff
@@ -34,6 +43,7 @@
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.Group;
 import org.eclipse.swt.widgets.Label;
+import org.eclipse.ui.PlatformUI;
```
[the remark this answers is not in the document]
Comment (adietish) [reply plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionComposite.java]: @robstryker dare to address please? simple enough.

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionComposite.java [hunk 2]
```diff
@@ -17,10 +17,19 @@
...
+import org.eclipse.core.databinding.observable.list.IObservableList;
+import org.eclipse.core.databinding.observable.list.WritableList;
+import org.eclipse.core.databinding.observable.value.IObservableValue;
+import org.eclipse.core.databinding.observable.value.WritableValue;
+import org.eclipse.core.databinding.validation.MultiValidator;
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.Status;
 import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jface.databinding.fieldassist.ControlDecorationSupport;
```
[the remark this answers is not in the document]
Comment (adietish) [reply plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionComposite.java]: @robstryker dare to address please? simple enough.

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionComposite.java [hunk 3]
```diff
@@ -17,10 +17,19 @@
 import java.util.Map;
 import java.util.stream.Collectors;
 
+import org.eclipse.core.databinding.DataBindingContext;
+import org.eclipse.core.databinding.observable.list.IObservableList;
+import org.eclipse.core.databinding.observable.list.WritableList;
+import org.eclipse.core.databinding.observable.value.IObservableValue;
+import org.eclipse.core.databinding.observable.value.WritableValue;
+import org.eclipse.core.databinding.validation.MultiValidator;
```
[the remark this answers is not in the document]
Comment (adietish) [reply plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionComposite.java]: @robstryker dare to address please? simple enough.

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionComposite.java [hunk 4]
```diff
@@ -17,10 +17,19 @@
 import java.util.Map;
 import java.util.stream.Collectors;
 
+import org.eclipse.core.databinding.DataBindingContext;
+import org.eclipse.core.databinding.observable.list.IObservableList;
+import org.eclipse.core.databinding.observable.list.WritableList;
+import org.eclipse.core.databinding.observable.value.IObservableValue;
+import org.eclipse.core.databinding.observable.value.WritableValue;
```
[the remark this answers is not in the document]
Comment (adietish) [reply plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionComposite.java]: @robstryker dare to address please? simple enough.

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionComposite.java [hunk 5]
```diff
@@ -17,10 +17,19 @@
 import java.util.Map;
 import java.util.stream.Collectors;
 
+import org.eclipse.core.databinding.DataBindingContext;
+import org.eclipse.core.databinding.observable.list.IObservableList;
+import org.eclipse.core.databinding.observable.list.WritableList;
+import org.eclipse.core.databinding.observable.value.IObservableValue;
```
[the remark this answers is not in the document]
Comment (adietish) [reply plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionComposite.java]: @robstryker dare to address please? simple enough.

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionComposite.java [hunk 6]
```diff
@@ -17,10 +17,19 @@
 import java.util.Map;
 import java.util.stream.Collectors;
 
+import org.eclipse.core.databinding.DataBindingContext;
+import org.eclipse.core.databinding.observable.list.IObservableList;
+import org.eclipse.core.databinding.observable.list.WritableList;
```
[the remark this answers is not in the document]
Comment (adietish) [reply plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionComposite.java]: @robstryker dare to address please? simple enough.

**Comment by adietish** [comment]
@robstryker dare to please remove the unused imports as reported by sonar?

**Comment by robstryker** [comment]
@adietish patch cleaned up. Please +1 it ;)

**Comment by robstryker** [comment]
testPR

**Pull request merged**